testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveViewsCompatibility.java

```diff
@@ -117,6 +117,66 @@ public void testIcebergHiveViewsCompatibility()
         }
     }
 
+    @Test(groups = {ICEBERG, STORAGE_FORMATS, HMS_ONLY})
+    public void testIcebergMaterializedViewHiveViewsCompatibility()
```

Comment:
This test succeeds without the above change. Please make sure that new tests fail without the change when fixing a bug.

Reply:
oh! Does `SHOW TABLES` not delegate to getView or some derivative? Would I need to query from information schema?

Reply:
SHOW TABLES presents a different information that information_schema.tables, so it may or may not be following same exec path

Reply:
Turns out, I'm struggling to trigger this. Something that might be making it hard for me to trigger this is that the getViews method for HiveMetadata swallows all "ViewTranslationErrors"

```
ImmutableMap.Builder<SchemaTableName, ConnectorViewDefinition> views = ImmutableMap.builder();
        for (SchemaTableName name : listViews(session, schemaName)) {
            try {
                getView(session, name).ifPresent(view -> views.put(name, view));
            }
            catch (TrinoException e) {
                if (e.getErrorCode().equals(HIVE_VIEW_TRANSLATION_ERROR.toErrorCode())) {
                    // Ignore hive views for which translation fails
                }
                else if (e.getErrorCode().equals(HIVE_INVALID_VIEW_DATA.toErrorCode())) {
                    // Ignore views that are not valid
                }
                else if (e.getErrorCode().equals(TABLE_NOT_FOUND.toErrorCode())) {
                    // Ignore view that was dropped during query execution (race condition)
                }
                else {
                    throw e;
                }
            }
        }
        return views.buildOrThrow();
  ``` 
  
And the condition that this change stops from throwing is one of the ones caught here: `HIVE_INVALID_VIEW_DATA`